### ISSUE TITLE
Openblas library not loaded

### ISSUE THREAD

**Issue by GertjanBisschop** [tskit-dev/msprime#629]
installation of msprime did not include openblas library

>>> import msprime
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/s1854903/miniconda3/envs/slm/lib/python3.6/site-packages/msprime/__init__.py", line 26, in <module>
    from _msprime import FORWARD  # NOQA
ImportError: dlopen(/Users/s1854903/miniconda3/envs/slm/lib/python3.6/site-packages/_msprime.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libopenblas.dylib
  Referenced from: /Users/s1854903/miniconda3/envs/slm/lib/python3.6/site-packages/_msprime.cpython-36m-darwin.so
  Reason: image not found

upon installing openblas the package loaded perfectly.

**Comment by jeromekelleher** [comment]
Thanks for letting me know about this issue @GertjanBisschop. I think I've hit something similar updating the recipes for conda-forge [here](https://github.com/conda-forge/msprime-feedstock/pull/14), but I'm having trouble tracking it down as I don't have access to a mac.

Would you mind getting as much detail as you can about your conda environment and pasting it in here please? If you could give the steps to reproduce the problem as well that'd be really helpful.

**Comment by molpopgen** [comment]
I'm having this happen on Linux, too. Travis CI tests for fwdpy11 that involve msprime are failing with this error.

**Comment by jeromekelleher** [comment]
This is an upstream packaging bug, so I'm going to close this issue. For reference, see https://github.com/conda-forge/msprime-feedstock/pull/18 and https://github.com/conda-forge/gsl-feedstock/issues/34

**Comment by gbradburd** [comment]
Hi Jerome,

I'm getting what seems like a related problem on Mac OS 10.13.6 with python 3.7.4.  Running `python3 -m pip install msprime` results in "Successfully installed msprime-0.7.1".  But, when I try `$ mspms` I get the following error:

Traceback (most recent call last):
  File "./mspms", line 6, in <module>
    from msprime.cli import mspms_main
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/msprime/__init__.py", line 49, in <module>
    from msprime.provenance import __version__
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/msprime/provenance.py", line 25, in <module>
    import _msprime
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/_msprime.cpython-37m-darwin.so, 2): Library not loaded: @rpath/libcblas.3.dylib
  Referenced from: /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/_msprime.cpython-37m-darwin.so
  Reason: image not found

FWIW, msprime was working fine before I updated to Python 3.7.  Any light you could shed would be much appreciated!

-Gideon

**Comment by jeromekelleher** [comment]
Looks like the problem here is that GSL can't be found at run-time. How have you installed GSL? Are you using conda or some other Python installation option? I'm afraid I can't help much here as I'm not a mac user...

**Comment by molpopgen** [comment]
OS X often doesn't have the environment set to include places like
/usr/local.  You may need to set DYLD_LIBRARY_PATH and/or set LDFLAGS to
hard-code in the rpath to GSL.

If the gsl-config command is not found, then you either don't have GSL
installed or it is installed in a location your environment doesn't know
about.

**Comment by gbradburd** [comment]
Thanks both for the prompt feedback!   I've got things working now, but I'm going to give a detailed response in case anyone else has this problem (or just me, in the future).  I'm a beginner with python, so apologies if I don't explain this clearly and w/ correct nomenclature.

@jeromekelleher, I had been using pip instead of the recommended conda install, and as Kevin pointed out, the problem was that somehow msprime wasn't correctly finding my dynamic libraries (gsl had been successfully installed with homebrew, and the dylib was in /usr/local/bin with the rest of my dynamic libs).  @molpopgen, I had tried to set DYLD_LIBRARY_PATH in my .bash_profile, but that ended up breaking lots of other things (e.g., pip), and I didn't feel like I had a solid enough handle on what I was doing to unbreak things.

I got into this pickle in the first place because I had been trying to write an executable python script for grabbing the tree-sequence output of a SLiM simulation, simulating mutations on it, and outputting the data as a .bcf file.  The script relied on msprime, tskit, and pyslim.  I had originally installed msprime via conda, and tskit and pyslim through pip, and the script worked fine in an interactive python environment.  However, when I tried to make the script executable (with `#!/miniconda3/bin/python3`) and ran it from a bash script, it failed because it was unable to import pyslim.  To fix this, I had uninstalled miniconda and tried to install all three packages through pip and make the script executable with `#!/usr/local/bin/python`.  That didn't work because msprime didn't run correctly (with the message listed in my original post).  Ultimately, I got everything working by installing 1) miniconda, 2) msprime through conda, 3) then pyslim and tskit with pip _after_ having installed miniconda.  Note (for my future self) that when I installed pyslim and tskit with pip _before_ installing miniconda, they didn't show up in my conda list, and the executable script wasn't be able to find them with `#!/miniconda3/bin/python3`. 

thanks again for the advice Jerome and Kevin!  Much appreciated.

-g
